addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -999,6 +999,30 @@ registry.anchorSlide = publicWidget.Widget.extend({
     },
 });
 
+// TODO: remove me in master.
+registry.anchorScrollTop = publicWidget.Widget.extend({
+    selector: 'a[href="#top"]',
+    events: {
+        'click': '_onTopClick',
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onTopClick: function (ev) {
+        if (!document.querySelector("#top")) {
+            // The header has been removed from the DOM and there is no element
+            // with an id set to "top". Directly scroll to the top of the
+            // document
+            dom.scrollTo(document.querySelector("#wrapwrap"));
```

Comment:
Any reason to not scroll always to the wrapwrap instead of top?

Reply:
Why not in stable, instead of the current diff (if we are sure that wrapwrap is always there, which seems to be the case as quickly looking, the `top` is a child of the wrapwrap) ?


It would not fix existing databases, but:
1. It's a niche scenario, people usually don't hide their header, and even if they do, they still have to use the scroll top feature
2. The support could very well just update the view should there be a ticket, and it should be easy to identify as it would reach the website bugfix worst case


All in all, I'd not bring more (always loaded) JS code for this, I don't think it's worth the "technical debt".

Note that even if we were to keep your JS stuff, you would have to at least add a comment in the view template next to that `<a href="top"` element to indicate that the href target is changed through JS when the header is hidden, otherwise it's really not ideal to have such JS code lost in our code base which no one will probably find when trying to understand stuff related to this option :)

Reply:
Yes it's a thing, but most likely through manual update of the view and not module update (they might tell to do it to the client but won't do it themselves (I hope).
Bugfixes are always a balance of many things. In this case it's between "clean code" vs "existing db won't receive the fix (as they won't -u)".
I am pretty sure that header less DBs with scroll top option is like 0.01% of our DBs, in such a case it's probably better to go for the clean code, knowing that only a few existing clients will face the issue and that new client will have the fix.
I am not even 100% sure my call is the best here, it's really a thin line but you got the point :+1: